### PR TITLE
Adding support for Xlant XL keyboard

### PR DESCRIPTION
**Pull request by ChrisGVE**
The change includes the keyboard layout, data driven with .json file, and a default layout as a starting point to customize a board.

## Description
The change include a new main folder `zlant_xl` for a single keyboard.

The keyboard is fully configured via a keyboard.json file. In addition a default layout is provided.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/zlant_xl/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define BASE 0
+#define FN 1
```
Comment (waffle87) [inline keyboards/zlant_xl/keymaps/default/keymap.c:20]:
```suggestion
enum layers {
    BASE,
    FN
};
```

File: keyboards/zlant_xl/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+		KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
+		_______, KC_LCTL, KC_LALT, KC_LGUI, KC_LSFT, KC_SPC,  KC_SPC,  MO(1),   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT),
+
+	[FN] = LAYOUT(
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+		_______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
+		_______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_HOME, KC_END,  _______,
+		_______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY),
```
Comment (waffle87) [inline keyboards/zlant_xl/keymaps/default/keymap.c:37]:
```suggestion
    [BASE] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
        _______, KC_LCTL, KC_LALT, KC_LGUI, KC_LSFT, KC_SPC,  KC_SPC,  MO(1),   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
    ),
    [FN] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
        _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_HOME, KC_END,  _______,
        _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
    )
```

File: keyboards/zlant_xl/readme.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# ZLANT XL
+
+![ZLANT XL](https://1upkeyboards.com/wp-content/uploads/2021/06/ZlantXL-scaled.jpeg)
```
Comment (waffle87) [inline keyboards/zlant_xl/readme.md:3]: This image should be rehosted on Imgur.

File: keyboards/zlant_xl/readme.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+_A unique 50% mechanical keyboard that uses a slanted ortholinear layout_
+
+- Keyboard Maintainer: [chrisgve](https://github.com/chrisgve)
+- Hardware Supported: Zlant XL w/Ziptyze's atmega32u2 powered controller [The Rune](https://1upkeyboards.com/shop/controllers/the-rune-by-ziptyze/)
+- Hardware Availability: [1UP Keyboards](https://1upkeyboards.com/shop/keyboard-kits/diy-40-kits/zlantxl-50-mechanical-keyboard-kit/)
```
Comment (waffle87) [inline keyboards/zlant_xl/readme.md:8]:
```suggestion
* Keyboard Maintainer: [chrisgve](https://github.com/chrisgve)
* Hardware Supported: Zlant XL w/Ziptyze's atmega32u2 powered controller [The Rune](https://1upkeyboards.com/shop/controllers/the-rune-by-ziptyze/)
* Hardware Availability: [1UP Keyboards](https://1upkeyboards.com/shop/keyboard-kits/diy-40-kits/zlantxl-50-mechanical-keyboard-kit/)
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
If this keyboard is made/sold by 1upkeyboards, it should probably live with the rest of the keyboards under `keyboards/1upkeyboards/...`.

**Comment by ChrisGVE** [comment]
Regarding the 1UpKeyboards folder, I noted that they sell the Zlant keyboard (https://1upkeyboards.com/shop/keyboard-kits/diy-40-kits/zlant-40-keyboard-pcb/), which lives in QMK in the main keyboards folder. That's why I kept the Zlant XL at the same level.
What is your suggestion?

**Comment by waffle87** [comment]
Ah, I suppose it looks like the ZlantXL was designed/sold outside of 1upkeyboards originally ([Reddit post](https://www.reddit.com/r/mechmarket/comments/mg6u84/bulk_zlant_xl_kits_powered_by_rune_a_5x12_uniform/)). So I suppose its current location is more appropriate.

**Review by waffle87 — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**